scicomp/ssh.rst

```diff
@@ -75,21 +111,79 @@ Copy public key to server
 
 In order to use your key-pair to login to Triton, you first need to securely copy the desired *public key* to the machine with ``ssh-copy-id``. The script will also add the key to the ``~/.ssh/authorized_keys`` file on the server. You will be prompted to enter your Aalto password to initiate the secure copy of the file to Triton.
 
-::
-
-    ssh-copy-id -i ~/.ssh/id_rsa_triton.pub login_name@triton.aalto.fi
+.. note:: Connecting from outside of the aalto network
+   
+   The following command works, if you are within the Aalto network or if you are connected via vpn. If you log in from outside Aalto, 
+   you first need to set up the key for the login server (e.g. kosh). This can be done by replacing ``login_name@triton.aalto.fi`` by  ``login_name@kosh.aalto.fi`` in the following command.
+   If you connect from outside aalto it is usefule to first set up the config file :ref:`as described below<example_config_for_ssh>`. 
+   Once this is done run the following command once with ``kosh`` instead of ``triton.aalto.fi`` followed by the same command with ``triton_via_kosh`` instead of ``triton.aalto.fi`` to 
+   transfer your public keys to both the firewall server kosh and triton.
+
+
+.. tabs::
+
+  .. group-tab:: Windows
+          
+     ::
+     
+         type %USERPROFILE%\.ssh\id_rsa_triton.pub | ssh login_name@triton.aalto.fi "cat >> .ssh/authorized_keys"      
+       
+	        
+  .. group-tab:: Linux
+     
+     ::
+      
+         ssh-copy-id -i ~/.ssh/id_rsa_triton.pub login_name@triton.aalto.fi
+        
+  .. group-tab:: Mac
+  
+     ::
+      
+         ssh-copy-id -i ~/.ssh/id_rsa_triton.pub login_name@triton.aalto.fi
+    
+
+    
 
 
 Login with SSH key
 -------------------
 
 To avoid having to type the decryption password, the *private key* it needs to be added to the ``ssh-agent`` with the command
 
-::
-
-    ssh-add ~/.ssh/id_rsa_triton
-
-Once the password is added, you can ssh into Triton as normal but will immediately be connected without any further prompts. If you are unsure whether a ``ssh-agent`` process is running on your machine, ``ps -C ssh-agent`` will tell you if there is. To start a new agent, use ``eval $(ssh-agent)``.
+.. tabs::
+
+  .. group-tab:: Windows
+        
+    	 You will need administrative permissions to be able to start a ssh-agent on your machine that can store and handle passwords. 
+    	     
+         1. Open *Services* from the start menu
+
+         2. Scroll down to *OpenSSH Authentication Agent* > *double click*
+
+         3. Change the *Startup type* to *Automatic (Delayed Start)*,
+            or anything that is not *Disabled*
+    
+         4. ``ssh-add %USERPROFILE%\.ssh\id_rsa_triton``
```

Comment:
With the default key name, is it automatically found?

Reply:
I never tried this with ssh-add. Anyone else?

Reply:
ssh-add with no arguments uses the default key location.  I only give a key name where I have a separate key with password, separate from the default passwordless one.